### ISSUE TITLE
The LSP is not initialized

### ISSUE THREAD

**Issue by breakingram** [tris203/rzls.nvim#2]
OS: Linux Ubuntu 24.04 LTS
NVIM: v0.10.0-dev-435dee7

After opening a .razor/.cshtml file, no LSP client get attached to the buffer.

Here is my `nvm-lspconfig` setup:
```lua
return { -- LSP Configuration & Plugins
  'neovim/nvim-lspconfig',
  dependencies = {
    -- Automatically install LSPs and related tools to stdpath for Neovim
    { 'williamboman/mason.nvim', config = true }, -- NOTE: Must be loaded before dependants
    'williamboman/mason-lspconfig.nvim',
    'WhoIsSethDaniel/mason-tool-installer.nvim',
    'tris203/rzls.nvim',
.
.
.
.
.


  local capabilities = vim.lsp.protocol.make_client_capabilities()
  capabilities = vim.tbl_deep_extend('force', capabilities, require('cmp_nvim_lsp').default_capabilities())

  require('rzls').setup {
    on_attach = function()
      vim.keymap.set('n', 'gd', vim.lsp.buf.definition, { buffer = 0 })
    end,
    capabilities = capabilities,
    path = '/home/ram/.vscode/extensions/ms-dotnettools.csharp-2.34.12-linux-x64/.razor/rzls',
  }

```

**Comment by tris203** [comment]
Do you get any messages @InfinytRam?

I havent tried with the VSCode bundled rzls and instead have used the one built from source in the razor repo

**Comment by InfinytRam** [comment]
Thanks for the quick response @tris203, 

This time i provided the path to `rzls` after cloning and build from source:
`dotnet/razor` source: https://github.com/dotnet/razor

Steps to build `dotnet/razor`:
1. Cloned the [dotnet/razor](https://github.com/dotnet/razor) repsitory
2. cd into razor
3. run `./build.sh` in terminal
4. Then found the `rzls` in the following path:
    a. `/home/ram/Documents/projects/dotnet/razor/razor/artifacts/LanguageServer/Debug/net8.0/linux-x64/rzls`

I provided the `rzls` path in 'rzls' setup in neovim:
```lua
    local capabilities = vim.lsp.protocol.make_client_capabilities()
    capabilities = vim.tbl_deep_extend('force', capabilities, require('cmp_nvim_lsp').default_capabilities())

    require('rzls').setup {
      on_attach = function()
        vim.keymap.set('n', 'gd', vim.lsp.buf.definition, { buffer = 0 })
      end,
      capabilities = capabilities,
      path = '/home/ram/Documents/projects/dotnet/razor/razor/artifacts/LanguageServer/Debug/net8.0/linux-x64/rzls',
    }

```
and unfortunately i did not get any messages when i opened .cshtml file. I also checked the `:messages` command in neovim. 

Here is my neovim config:
https://github.com/InfinytRam/mynvim

Happy to provide additional information.

**Comment by tris203** [comment]
Can you try and a .razor file?

Ps, this plugin is still very much under development

Only a very small amount of methods work and a lot of debug info is still visible

**Comment by nomis51** [comment]
I've tried with both VSCode `rzls` and `dotnet/razor` `rzls`, both giving those errors upon opening a razor file

![image](https://github.com/tris203/rzls.nvim/assets/25111613/f33c9559-ffdd-4b91-ae7e-e2ac1eb39eb3)

It then says "Razor LSP attached" and I can see "rzls" on the task bar, but it's a bit broken. Some warnings show off, but that's pretty much it.

![image](https://github.com/tris203/rzls.nvim/assets/25111613/1665aeb6-0cac-4d53-952b-f214ba3669e1)

And as soon as I input something in the file, it get those two above errors back to back on every keystroke basically.

Had to change the `cmd path` of the plugin to add `.exe` at the end for it to works on Windows.

![image](https://github.com/tris203/rzls.nvim/assets/25111613/c3c3a476-254c-454c-832e-bbee387207c2)

And here's my setup

![image](https://github.com/tris203/rzls.nvim/assets/25111613/ec611ab7-3187-4918-a25b-6c980447d982)

**Comment by tris203** [comment]
> I've tried with both VSCode `rzls` and `dotnet/razor` `rzls`, both giving those errors upon opening a razor file
Do you see it open the projected virtual buffers?

**Comment by nomis51** [comment]
> > I've tried with both VSCode `rzls` and `dotnet/razor` `rzls`, both giving those errors upon opening a razor file
> > Do you see it open the projected virtual buffers?

I think so

![image](https://github.com/tris203/rzls.nvim/assets/25111613/5dc8a660-a01e-4a11-a65c-7481760c0d87)

**Comment by tris203** [comment]
> > > I've tried with both VSCode `rzls` and `dotnet/razor` `rzls`, both giving those errors upon opening a razor file
> > > Do you see it open the projected virtual buffers?
> 
> I think so
> 
> ![image](https://private-user-images.githubusercontent.com/25111613/344491423-5dc8a660-a01e-4a11-a65c-7481760c0d87.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3NzgwNzEsIm5iZiI6MTcxOTc3Nzc3MSwicGF0aCI6Ii8yNTExMTYxMy8zNDQ0OTE0MjMtNWRjOGE2NjAtYTAxZS00YTExLWE2NWMtNzQ4MTc2MGMwZDg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjMwVDIwMDI1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiYjRiZmZkM2RhZDk1NzgxMDI2NGE2NWJlYWI3MGE4YzE0ODYzZWMwMDVjNTliYjJkZDBmYmRiNDg3MDBhNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bi5F8eDA18K9RUtv0kssXc6vNOa-xRff-8FStaCkrp4)

You will also need `roslyn.nvim`. Which might be the cause of the issue
https://github.com/jmederosalvarado/roslyn.nvim

**Comment by InfinytRam** [comment]
Hey @tris203, 

I managed to progress a little bit on Linux Ubuntu side and now i'm finally getting error message feedback when I open .razor file.

My mistake was, for the path I was adding the file  `rzls` instead of the path containing `rzls`.

For example I changed the path from this:
`path = '/home/ram/Documents/projects/dotnet/razor/razor/artifacts/LanguageServer/Debug/net8.0/linux-x64/rzls',
`

To this:
`path = '/home/ram/Documents/projects/dotnet/razor/razor/artifacts/LanguageServer/Debug/net8.0/linux-x64/',`

Then When I opened a Razor file, I get the following messages (also provided error msg in text below):
[Screencast from 2024-06-30 17-54-11.webm](https://github.com/tris203/rzls.nvim/assets/38049078/ef046efe-7863-4cd0-9124-004a49874b00)

Also when I try to save the .razor file, it creates a `*.razor__virtual.html` file like this:
![Screenshot from 2024-06-30 18-01-30](https://github.com/tris203/rzls.nvim/assets/38049078/8dce7627-ab1f-4857-83a3-7b6cd80d9d3e)


Error messages in the video in text:

First error message:
``` lua
Error executing vim.schedule lua callback: /usr/local/share/nvim/runtime/lua/vim/lsp/diagnostic.lua:111: attempt to index field 'range' (a nil value)
stack traceback:
	/usr/local/share/nvim/runtime/lua/vim/lsp/diagnostic.lua:111: in function </usr/local/share/nvim/runtime/lua/vim/lsp/diagnostic.lua:110>
	vim/shared.lua: in function 'diagnostic_lsp_to_vim'
	/usr/local/share/nvim/runtime/lua/vim/lsp/diagnostic.lua:260: in function 'handle_diagnostics'
	/usr/local/share/nvim/runtime/lua/vim/lsp/diagnostic.lua:293: in function 'handler'
	/usr/local/share/nvim/runtime/lua/vim/lsp/client.lua:1005: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>

```

Second error message:
``` lua
   Error  05:57:13 PM msg_show.lua_error Error executing vim.schedule lua callback: UnhandledPromiseRejection with the reason:
RPC[Error] code_name = unknown, code = -32004, message = "Deserializing JSON-RPC result to type RazorFoldingRangeResponse failed with JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'Microsoft.CodeAnalysis.Razor.Protocol.Folding.RazorFoldingRangeResponse' because the type requires a JSON object (e.g. {\"name\":\"value\"}) to deserialize correctly.\
To fix this error either change the JSON to a JSON object (e.g. {\"name\":\"value\"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.\
Path 'result'." data = {
  AssemblyName = "Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
  ClassName = "Newtonsoft.Json.JsonSerializationException",
  HResult = -2146233088,
  InnerException = {
    AssemblyName = "Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
    ClassName = "Newtonsoft.Json.JsonSerializationException",
    HResult = -2146233088,
    Message = "Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'Microsoft.CodeAnalysis.Razor.Protocol.Folding.RazorFoldingRangeResponse' because the type requires a JSON object (e.g. {\"name\":\"value\"}) to deserialize correctly.\nTo fix this error either change the JSON to a JSON object (e.g. {\"name\":\"value\"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.\nPath 'result'.",
    RemoteStackIndex = 0,
    Source = "Newtonsoft.Json",
    StackTraceString = "   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureArrayContract(JsonReader reader, Type objectType, JsonContract contract)\n   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)\n   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)\n   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)\n   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)\n   at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)\n   at Newtonsoft.Json.Linq.JToken.ToObject(Type objectType, JsonSerializer jsonSerializer)\n   at Newtonsoft.Json.Linq.JToken.ToObject[T](JsonSerializer jsonSerializer)\n   at StreamJsonRpc.JsonMessageFormatter.JsonRpcResult.GetResult[T]()"
  },
  Message = "Deserializing JSON-RPC result to type RazorFoldingRangeResponse failed with JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'Microsoft.CodeAnalysis.Razor.Protocol.Folding.RazorFoldingRangeResponse' because the type requires a JSON object (e.g. {\"name\":\"value\"}) to deserialize correctly.\nTo fix this error either change the JSON to a JSON object (e.g. {\"name\":\"value\"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.\nPath 'result'.",
  RemoteStackIndex = 0,
  Source = "StreamJsonRpc",
  StackTraceString = "   at StreamJsonRpc.JsonMessageFormatter.JsonRpcResult.GetResult[T]()\n   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)\n   at Microsoft.AspNetCore.Razor.LanguageServer.ClientConnection.SendRequestAsync[TParams,TResponse](String method, TParams params, CancellationToken cancellationToken) in /home/ram/Documents/projects/dotnet/razor/razor/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ClientConnection.cs:line 27\n   at Microsoft.AspNetCore.Razor.LanguageServer.Folding.FoldingRangeEndpoint.HandleCoreAsync(RazorFoldingRangeRequestParam requestParams, DocumentContext documentContext, CancellationToken cancellationToken) in /home/ram/Documents/projects/dotnet/razor/razor/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs:line 85\n   at Microsoft.AspNetCore.Razor.LanguageServer.Folding.FoldingRangeEndpoint.HandleRequestAsync(FoldingRangeParams params, RazorRequestContext requestContext, CancellationToken cancellationToken) in /home/ram/Documents/projects/dotnet/razor/razor/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs:line 70\n   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`3.StartRequestAsync(TRequestContext context, IMethodHandler handler, CancellationToken cancellationToken) in /home/ram/.nuget/packages/microsoft.commonlanguageserverprotocol.framework/4.11.0-3.24303.3/contentFiles/cs/netstandard2.0/QueueItem.cs:line 136\n   at Microsoft.CommonLanguageServerProtocol.Framework.AbstractLanguageServer`1.DelegatingEntryPoint.InvokeAsync(MethodInfo methodInfo, IRequestExecutionQueue`1 queue, Object requestObject, String language, ILspServices lspServices, CancellationToken cancellationToken) in /home/ram/.nuget/packages/microsoft.commonlanguageserverprotocol.framework/4.11.0-3.24303.3/contentFiles/cs/netstandard2.0/AbstractLanguageServer.cs:line 238\n   at Microsoft.CommonLanguageServerProtocol.Framework.NewtonsoftLanguageServer`1.NewtonsoftDelegatingEntryPoint.ExecuteRequestAsync(JToken request, CancellationToken cancellationToken) in /home/ram/.nuget/packages/microsoft.commonlanguageserverprotocol.framework/4.11.0-3.24303.3/contentFiles/cs/netstandard2.0/NewtonsoftLanguageServer.cs:line 69"
}
   Error  05:57:26 PM msg_show.lua_error   w Error executing vim.schedule lua callback: UnhandledPromiseRejection with the reason:
RPC[Error] code_name = unknown, code = -32004, message = "Deserializing JSON-RPC result to type RazorFoldingRangeResponse failed with JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'Microsoft.CodeAnalysis.Razor.Protocol.Folding.RazorFoldingRangeResponse' because the type requires a JSON object (e.g. {\"name\":\"value\"}) to deserialize correctly.\
To fix this error either change the JSON to a JSON object (e.g. {\"name\":\"value\"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.\
Path 'result'." data = {
  AssemblyName = "Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
  ClassName = "Newtonsoft.Json.JsonSerializationException",
  HResult = -2146233088,
  InnerException = {
    AssemblyName = "Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed",
    ClassName = "Newtonsoft.Json.JsonSerializationException",
    HResult = -2146233088,
    Message = "Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'Microsoft.CodeAnalysis.Razor.Protocol.Folding.RazorFoldingRangeResponse' because the type requires a JSON object (e.g. {\"name\":\"value\"}) to deserialize correctly.\nTo fix this error either change the JSON to a JSON object (e.g. {\"name\":\"value\"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.\nPath 'result'.",
    RemoteStackIndex = 0,
    Source = "Newtonsoft.Json",
    StackTraceString = "   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureArrayContract(JsonReader reader, Type objectType, JsonContract contract)\n   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)\n   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)\n   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)\n   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)\n   at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)\n   at Newtonsoft.Json.Linq.JToken.ToObject(Type objectType, JsonSerializer jsonSerializer)\n   at Newtonsoft.Json.Linq.JToken.ToObject[T](JsonSerializer jsonSerializer)\n   at StreamJsonRpc.JsonMessageFormatter.JsonRpcResult.GetResult[T]()"
  },
  Message = "Deserializing JSON-RPC result to type RazorFoldingRangeResponse failed with JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'Microsoft.CodeAnalysis.Razor.Protocol.Folding.RazorFoldingRangeResponse' because the type requires a JSON object (e.g. {\"name\":\"value\"}) to deserialize correctly.\nTo fix this error either change the JSON to a JSON object (e.g. {\"name\":\"value\"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.\nPath 'result'.",
  RemoteStackIndex = 0,
  Source = "StreamJsonRpc",
  StackTraceString = "   at StreamJsonRpc.JsonMessageFormatter.JsonRpcResult.GetResult[T]()\n   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)\n   at Microsoft.AspNetCore.Razor.LanguageServer.ClientConnection.SendRequestAsync[TParams,TResponse](String method, TParams params, CancellationToken cancellationToken) in /home/ram/Documents/projects/dotnet/razor/razor/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ClientConnection.cs:line 27\n   at Microsoft.AspNetCore.Razor.LanguageServer.Folding.FoldingRangeEndpoint.HandleCoreAsync(RazorFoldingRangeRequestParam requestParams, DocumentContext documentContext, CancellationToken cancellationToken) in /home/ram/Documents/projects/dotnet/razor/razor/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs:line 85\n   at Microsoft.AspNetCore.Razor.LanguageServer.Folding.FoldingRangeEndpoint.HandleRequestAsync(FoldingRangeParams params, RazorRequestContext requestContext, CancellationToken cancellationToken) in /home/ram/Documents/projects/dotnet/razor/razor/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs:line 70\n   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`3.StartRequestAsync(TRequestContext context, IMethodHandler handler, CancellationToken cancellationToken) in /home/ram/.nuget/packages/microsoft.commonlanguageserverprotocol.framework/4.11.0-3.24303.3/contentFiles/cs/netstandard2.0/QueueItem.cs:line 136\n   at Microsoft.CommonLanguageServerProtocol.Framework.AbstractLanguageServer`1.DelegatingEntryPoint.InvokeAsync(MethodInfo methodInfo, IRequestExecutionQueue`1 queue, Object requestObject, String language, ILspServices lspServices, CancellationToken cancellationToken) in /home/ram/.nuget/packages/microsoft.commonlanguageserverprotocol.framework/4.11.0-3.24303.3/contentFiles/cs/netstandard2.0/AbstractLanguageServer.cs:line 238\n   at Microsoft.CommonLanguageServerProtocol.Framework.NewtonsoftLanguageServer`1.NewtonsoftDelegatingEntryPoint.ExecuteRequestAsync(JToken request, CancellationToken cancellationToken) in /home/ram/.nuget/packages/microsoft.commonlanguageserverprotocol.framework/4.11.0-3.24303.3/contentFiles/cs/netstandard2.0/NewtonsoftLanguageServer.cs:line 69"
}

```

**Comment by InfinytRam** [comment]
@tris203 

![Screenshot from 2024-10-04 18-43-41](https://github.com/user-attachments/assets/16625380-7dbb-4bd3-8803-5eb8e980bbf1)


Getting the same error as @nomis51 when open a `.razor` file.
```lua
Creating virtual buffers for /home/ram/Documents/projects/dotnet/IMS/IMS.WebApp/Components/App.razor
Error executing vim.schedule lua callback: ...hare/nvim/lazy/rzls.nvim/lua/rzls/documentstore/init.lua:59: attempt to index a nil value
stack traceback:
	...hare/nvim/lazy/rzls.nvim/lua/rzls/documentstore/init.lua:59: in function 'update_html_vbuf'
	...cal/share/nvim/lazy/rzls.nvim/lua/rzls/handlers/init.lua:38: in function 'handler'
	/usr/local/share/nvim/runtime/lua/vim/lsp/client.lua:1005: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
Error executing vim.schedule lua callback: ...hare/nvim/lazy/rzls.nvim/lua/rzls/documentstore/init.lua:21: attempt to index a nil value
stack traceback:
	...hare/nvim/lazy/rzls.nvim/lua/rzls/documentstore/init.lua:21: in function 'update_csharp_vbuf'
	...cal/share/nvim/lazy/rzls.nvim/lua/rzls/handlers/init.lua:34: in function 'handler'
	/usr/local/share/nvim/runtime/lua/vim/lsp/client.lua:1005: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
Error executing vim.schedule lua callback: ...hare/nvim/lazy/rzls.nvim/lua/rzls/documentstore/init.lua:59: attempt to index a nil value

```


my config setup:
```lua
local capabilities = vim.lsp.protocol.make_client_capabilities()
capabilities = vim.tbl_deep_extend('force', capabilities, require('cmp_nvim_lsp').default_capabilities())

local rzls_path = '.........../razor/artifacts/LanguageServer/Debug/net8.0/linux-x64/'

require('rzls').setup {
  on_attach = function() end,
  capabilities = capabilities,
  path = rzls_path,
}

```

**Comment by tris203** [comment]
Can you guys try again. There have been a lot of updates to the workings of this

**Comment by nomis51** [comment]
I finally was able to make it work with your latest work (on native Windows). Can't say it's an easy ride yet, but it's getting there, great work! 

I installed roslyn manually (but I believe it would work with mason aswell) and installed rzls with mason.

I had to path manually to the `rzls.cmd` file located in the nvim-data folder (where neovim packages/data are stored on Windows) for rzls to work. I used your example roslyn config.

And finally it does work, both rzls and roslyn hook up when required. I didn't experience the problem of "open a .cs file first breaks rzls". For me no matter in what other I open files at first, it seems to work (razor then cs, cs then razor, cshtml then cs, etc.)

Although I had one error spamming 5 times as I first open a razor file (but it didn't seem to break anything afaik)

```
Error executing luv callback:

 ...ocal/nvim-data/lazy/rzls.nvim/lua/rzls/documentstore.lua:38: E5560: Vimscript function must not be called in a lua loop callback stack traceback: 
 
 [C]: in function 'uri_to_bufnr' ...ocal/nvim-data/lazy/rzls.nvim/lua/rzls/documentstore.lua:38: in function 'register_vbufs_by_path' 
 
 ...AppData/Local/nvim-data/lazy/rzls.nvim/lua/rzls/init.lua:180: in function <...AppData/Local/nvim-data/lazy/rzls.nvim/lua/rzls/init.lua:175>
```

Also, it might simply be WIP as of now, but I'm pointing it out : local @code blocks work, LSP provide completion, but not for code-behind

![image](https://github.com/user-attachments/assets/24654027-ec2d-444b-9de0-3ac4a007b449)

But if I use an explicit code-behind class, it does work

![image](https://github.com/user-attachments/assets/851d630e-e1dd-4bb1-add6-fef1dfa95d6c)
